File: 0080-remove-duplicates-from-sorted-array-ii/0080-remove-duplicates-from-sorted-array-ii.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var removeDuplicates = function(nums) {
    let p1 = 1;
    let mark = false;

    for (let i = 0; i < nums.length - 1; i++) {
        if (nums[i] === nums[i+1]) {
            if (mark === true) continue;
            else {
                nums[p1] = nums[i+1];
                p1++;
                mark = true;
            };
        } else {
            nums[p1] = nums[i+1];
            p1++;
            mark = false;
        };
    };
    
    return p1;
};

File: 0080-remove-duplicates-from-sorted-array-ii/0080-remove-duplicates-from-sorted-array-ii.py
class Solution:
    def removeDuplicates(self, nums: List[int]) -> int:
        mark = False
        p1 = 1
        for i in range(len(nums)-1):
            if nums[i] == nums[i+1] and mark == True:
                continue
            elif nums[i] == nums[i+1] and mark == False:
                nums[p1] = nums[i+1]
                p1+=1
                mark = True
            else:
                nums[p1] = nums[i+1]
                p1+=1
                mark = False
        return p1

